File: dich_chuyen_quan_hau.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int x, y, u, v;
        cin >> x >> y >> u >> v;

        // Tính khoảng cách theo hàng ngang và cột dọc
        int hor = abs(x - u);
        int ver = abs(y - v);

        if (hor == 0 && ver == 0) {
            cout << 0 << endl;
        } else if (hor == ver || hor == 0 || ver == 0) {
            // Nếu hậu có thể di chuyển theo đường chéo hoặc theo hàng ngang/cột dọc
            cout << 1 << endl;
        } else {
            // Trường hợp còn lại, hậu cần di chuyển ít nhất 2 lần để đi từ (x, y) đến (u, v)
            cout << 2 << endl;
        }
    }

    return 0;
}

File: dich_chuyen_quan_hau.java
import java.util.Scanner;

public class dich_chuyen_quan_hau {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();

        while (t-- > 0) {
            int x = scanner.nextInt();
            int y = scanner.nextInt();
            int u = scanner.nextInt();
            int v = scanner.nextInt();

            // Calculate horizontal and vertical distances
            int hor = Math.abs(x - u);
            int ver = Math.abs(y - v);

            if (hor == 0 && ver == 0) {
                System.out.println(0);
            } else if (hor == ver || hor == 0 || ver == 0) {
                System.out.println(1);
            } else {
                System.out.println(2);
            }
        }
        scanner.close();
    }
}
